ee/kernel/src/timer.c

```diff
@@ -13,7 +13,147 @@
  * Some routines to do some timer work
  */
 
-#include <tamtypes.h>
+#include <timer.h>
+#include <kernel.h>
+
+#ifdef TIME_USE_T0
+#define INTC_TIM       INTC_TIM0
+#define T_COUNT        T0_COUNT
+#define T_MODE         T0_MODE
+#define T_COMP         T0_COMP
+#else
+#define INTC_TIM       INTC_TIM1
+#define T_COUNT        T1_COUNT
+#define T_MODE         T1_MODE
+#define T_COMP         T1_COMP
+#endif
+
+#ifdef F___time_internals
+s32  __time_intr_overflow_id = -1;
+u64  __time_intr_overflow_count = 0;
+#else
+extern s32  __time_intr_overflow_id;
+extern u64  __time_intr_overflow_count;
+#endif
+
+#ifdef F_InitTimer
+static s32 intrOverflow(int ca)
+{
+   (void)ca;
+
+   __time_intr_overflow_count++;
+
+   // A write to the overflow flag will clear the overflow flag
+   // ---------------------------------------------------------
+   *T_MODE |= (1 << 11);
+
+   ExitHandler();
+   return -1;
+}
+
+/** Init the timer
+ *
+ *
+ * Start the interruption handler for getting the system time
+ * This function should be called at beginning of your program
+ */
+void InitTimer(void)
```

Comment:
The naming of this function pair (InitTimer / EndTimer) is misleading.

What about StartTimerSystemTime and StopTimerSystemTime (as in @uyjulian PR) ?

Reply:
Yes! Doing it.